src/Mvc/Mvc.Testing.Tasks/src/GenerateMvcTestManifestTask.cs

```diff
@@ -35,12 +40,27 @@ public override bool Execute()
         {
             using var fileStream = File.Create(ManifestPath);
             var output = new Dictionary<string, string>();
-
             foreach (var project in Projects)
             {
                 var contentRoot = project.GetMetadata("ContentRoot");
                 var assemblyName = project.GetMetadata("Identity");
-                output[assemblyName] = contentRoot;
+
+                // This is only set when publishing
+                if (string.IsNullOrEmpty(PathToCopyDeps))
+                {
+                    output[assemblyName] = contentRoot;
+                }
+                else
+                {
+                    // When publishing content root is always the BaseDirectory
+                    output[assemblyName] = "~";
+                    var depsFile = project.GetMetadata("DepsFile");
+                    Log.LogMessage("Looking for " + depsFile + ": "+ File.Exists(depsFile));
+                    if (File.Exists(depsFile))
+                    {
+                        File.Copy(depsFile, Path.Combine(PathToCopyDeps, Path.GetFileName(depsFile)));
```

Comment:
One small nit is that it might be better to do this within a target so that we can leverage `<Copy` which will do retries and so on.

I think it might be easier to just add these files to the resolvedfilestopublish itemgroup and that way it reuses the same standard insfrastructure.

Reply:
Cool, yeah that looks better than doing it in the task, updated with doing it in the targets file